Allow status command to include optional arbitrary JSON fields

andreastt opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/w3c/webdriver/334)
<!-- Reviewable:end -->

jgraham commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[webdriver-spec.html, line 1995 at r1](https://reviewable.io:443/reviews/w3c/webdriver/334#-KSfXb3YjtdqrAH3mss_:-KSfXb3ZovcL0MF-l9y0:b4oc4kx) ([raw file](https://github.com/w3c/webdriver/blob/958b4711a7313d1c91c3b6e87e137c5b0b5cd6c7/webdriver-spec.html#L1995)):*
> ```HTML
>  <li><p>Let <var>body</var> be a new JSON <a>Object</a>.
> 
>  <li><p>Optionally run implemention-specific steps
> ```

I think this is generally true of any command, especially for intermediary nodes (e.g. for authentication information)

---

*[webdriver-spec.html, line 2010 at r1](https://reviewable.io:443/reviews/w3c/webdriver/334#-KSfXpwhc4r_BVg-1WnK:-KSfXpwhc4r_BVg-1WnL:brnxm3a) ([raw file](https://github.com/w3c/webdriver/blob/958b4711a7313d1c91c3b6e87e137c5b0b5cd6c7/webdriver-spec.html#L2010)):*
> ```HTML
>     <p>Otherwise true.
>  
>    <dt><a>endpoint node</a>
> ```

I don't think this is really clearer than the vauge handwavy text since you have to say "let it be true (oh except it might not be really).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/webdriver/334)*
<!-- Sent from Reviewable.io -->

andreastt commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[webdriver-spec.html, line 1995 at r1](https://reviewable.io:443/reviews/w3c/webdriver/334#-KSfXb3YjtdqrAH3mss_:-KSfbzM3o6IqpIbCQIvl:b-xaht9n) ([raw file](https://github.com/w3c/webdriver/blob/958b4711a7313d1c91c3b6e87e137c5b0b5cd6c7/webdriver-spec.html#L1995)):*
<details><summary><i>Previously, jgraham wrote…</i></summary>
> I think this is generally true of any command, especially for intermediary nodes (e.g. for authentication information)
</details>
The point is that it needs to happen before the `ready` and `message` fields overwrite whatever extra data the implementation throws in there.

I can reword it in a less step-by-step way to say that these two fields are reserved, but in this particular case I think it’s worth pointing out that we fully expect more data to be added to the body.

---

*[webdriver-spec.html, line 2010 at r1](https://reviewable.io:443/reviews/w3c/webdriver/334#-KSfXpwhc4r_BVg-1WnK:-KSfctdK9kMfv7OGkgGT:b-9lal51) ([raw file](https://github.com/w3c/webdriver/blob/958b4711a7313d1c91c3b6e87e137c5b0b5cd6c7/webdriver-spec.html#L2010)):*
<details><summary><i>Previously, jgraham wrote…</i></summary>
> I don't think this is really clearer than the vauge handwavy text since you have to say "let it be true (oh except it might not be really).
</details>
I’m not sure what you’re referring to. The text for the endpoint node case or the intermediary node?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/webdriver/334)*
<!-- Sent from Reviewable.io -->

shs96c commented on webdriver-spec.html:
```diff
@@ -1968,42 +1968,72 @@
+ that a new session request will succeed.
+
+<p>Implementations may optionally include
+ arbitrary other meta information as part of the body,
```
Nit: No need for "arbitrary" here, as you've already said that this is the case above.

shs96c commented on webdriver-spec.html:
```diff
@@ -1968,42 +1968,72 @@
+
+    <p>Otherwise true.
+
+    <p>However, there may be further
```
Technically never reached, and incredibly hand-wavey. It would be better to default `readiness state` to `true`, and then let the unhandled case be dealt with implicitly.

andreastt replied:
This is a reasonable point. I believe I have addressed it now.

shs96c commented on webdriver-spec.html:
```diff
@@ -1968,42 +1968,72 @@
-  name <code>"ready"</code> and value <var>body</var>.
+ <li><p>Let <var>readiness state message</var>
+  be an implementation-defined string
+  explaining the value given to <var>readiness state</var>.
```
Indicate that the empty string is a legitimate response, but is only advised for when `readiness state` is `true`.

AutomatedTester commented:
@andreastt ping to finish up PR (when you have time)

andreastt commented:
@AutomatedTester Addressed code review issues. Removed some of the unnecessary stuff pointed out by jgraham and a logic inconsistency pointed out by simonstewart.
